src/libDirectoryService/ShardingPreProcessing.cpp

```diff
@@ -345,9 +287,9 @@ bool DirectoryService::RunConsensusOnShardingWhenDSPrimary()
     vector<unsigned char> sharding_structure;
 
     ComputeSharding();
-    SerializeShardingStructure(sharding_structure);
 
-    unsigned int txn_sharing_offset = sharding_structure.size();
+    unsigned int txn_sharing_offset
+        = ShardingStructure::Serialize(m_shards, sharding_structure, 0);
```

Comment:
It seems logic change here because ShardingStructure::Serialize() will return offset instead of sharding_structure.size().
Is it bug fix?

Reply:
In original code, offset was also expected (hence the name txn_sharing_offset), not size.
It just so happened that the offset was the same as the size.